src/TestSuite/TestCase.php

```diff
@@ -804,4 +804,25 @@ public static function setAppNamespace($appNamespace = 'TestApp')
     {
         Configure::write('App.namespace', $appNamespace);
     }
+
+    /**
+     * Gets fixtures.
+     *
+     * @return array
+     */
+    public function getFixtures()
+    {
+        if (!isset($this->fixtures)) {
+            return [];
+        }
+        if (is_string($this->fixtures)) {
+            deprecationWarning(
+                'Setting fixtures as string is deprecated and will be removed in 4.0.' .
+                ' Set TestCase::$fixtures as array instead.'
```

Comment:
Using `TestCase::$fixtures` would still lead to the memory leak. I would recommend to set fixtures in `getFixtures()` method like in `TableTest::getFixtures()` below without initiating `$fixtures` property.

Reply:
The memory leak figures you mentioned [here](https://github.com/cakephp/cakephp/issues/13583#issuecomment-526929833 are not really noteworthy TBH. Most users who actually have such large number of tests would have GBs worthy of memory at their disposable and wouldn't care about such a small memory bump. So smaller projects the memory bump would be insignificant.

So for majority of the users the convenience of setting the property is good enough.

Reply:
I may agree that the memory usage improvement will not be noticeable in smaller projects. But in larger projects where many tests have 20-30 fixtures it may easily reach 10% of memory improvement. It is also worthy to mention that memory leaks have greater effect on inspections when the code coverage report is generated. 

I do not insist here but in general the usage of public properties is discouraged.

Reply:
You can a note to the 3.next docs suggesting devs with large projects to avoid using the property. There's no point directing all users to do extra work by using the method, when the issue affects only a small minority.

With the addition of the getter method `TestCase::$fixtures` will be made protected in 4.x. We have already done that for most public properties.